rust/lance-index/src/scalar/inverted/builder.rs

```diff
@@ -324,6 +348,25 @@ mod tests {
         assert!(row_ids.contains(0));
         assert!(row_ids.contains(1));
         assert!(row_ids.contains(3));
+
+        // test phrase query
+        let row_ids = invert_index
+            .search(&SargableQuery::FullTextSearch(
+                FullTextSearchQuery::new("\"lance database\"".to_owned()).limit(Some(3)),
+            ))
+            .await
+            .unwrap();
+        assert_eq!(row_ids.len(), Some(2));
+        assert!(row_ids.contains(0));
+        assert!(row_ids.contains(1));
+
+        let row_ids = invert_index
+            .search(&SargableQuery::FullTextSearch(
+                FullTextSearchQuery::new("\"database lance\"".to_owned()).limit(Some(3)),
+            ))
+            .await
+            .unwrap();
+        assert_eq!(row_ids.len(), Some(0));
```

Comment:
Can you add one more case for `database lance` (no phrase query) and make sure it returns 2 results?

Reply:
good point! just added a test that query "lance database" and asserts there should be 4 documents hit (because it's OR)